### DISCUSSION TITLE
project post - can someone help me with my subscription message limit? Will pay. :)

### DISCUSSION THREAD

**Discussion by MoonstoneApp** [community/community#64050] [Programming Help]
Tools I've been using are visual studio code, github, render, revenuecat, flutterflow, firebase.
### **Desired outcomes:**
**Message Limits:**
- Free: 2 requests / month. The “month” starts from the date the user created an account and resets every 30 days.
- Premium: 20 requests / day.
- Monthly plan: month resets every 30 days.
- Annual plan: resets every 365 days.
- When user hits limit, they get a message back telling them to upgrade ideally with a button to upgrade. [I have something like this in place for limits minus the button part]

**2. There is a counter that lets the user know how many messages they have remaining on profile page**
For a free user, it should say X requests remaining for the month
For a paid user, it should say X requests remaining for the day

**Comment by tuves** [comment]
Hi @MoonstoneApp, we’re glad you’re here 😀

Thanks for sharing your project with the community. Please keep in mind that we currently do not allow advertising in the Community, per the [guidelines](https://github.com/orgs/community/discussions/51384). We want to make sure there is space for users to ask questions without overwhelming them with other conversations. Thank you for helping us maintain a productive and tidy community for all our members.
